index.js

```diff
@@ -0,0 +1,39 @@
+import React, { cloneElement, Children } from 'react';
+import { makeDecorator } from '@storybook/addons';
+
+const createFieldsObject = fields => {
+  if (!fields) {
+    return {};
+  }
+  return Object.keys(fields).reduce((acc, key) => {
+    const value = fields[key];
+    const isStringValue = typeof value === 'string';
+    acc[key] = isStringValue ? { value } : value;
+    return acc;
+  }, {});
+};
+
+const StoryContainer = ({ storyProps, children }) => {
+  const childComponents = Children.map(children, child => {
+    return cloneElement(child, {
+      ...storyProps,
+      fields: createFieldsObject(storyProps.fields)
+    });
+  });
+  return <>
+    {childComponents}
+  </>;
```

Comment:
[question] Can you just return `childComponents` here? It's an array, which I think is acceptable to nest within your <StoryContainer>